quic/s2n-quic/src/provider/endpoint_limits.rs

```diff
@@ -88,10 +214,28 @@ pub mod default {
     /// Default implementation for the Limits
     impl super::Limiter for Limits {
         fn on_connection_attempt(&mut self, info: &ConnectionAttempt) -> Outcome {
-            if s2n_quic_core::path::remote_port_blocked(info.remote_address.port()) {
+            let remote_port = info.remote_address.port();
+            if s2n_quic_core::path::remote_port_blocked(remote_port) {
                 return Outcome::drop();
             }
 
+            if s2n_quic_core::path::remote_port_throttled(remote_port) {
+                let port_index = s2n_quic_core::path::remote_port_throttled_index(remote_port);
+                match (self.rate_limiter, port_index) {
+                    (Some(rate_limiter), Some(port_index)) => {
+                        let mut rate_limiter = rate_limiter[port_index];
```

Comment:
so this is actually a bug... you're copying the rate limiter out of the array when you should be borrowing it instead. I would remove the `derive(Copy)` for the rate limiter.

Reply:
I added it because 
```
rror[E0204]: the trait `Copy` may not be implemented for this type
   --> quic/s2n-quic/src/provider/endpoint_limits.rs:201:21
    |
201 |     #[derive(Clone, Copy, Debug)]
    |                     ^^^^
...
205 |         rate_limiter: Option<[BasicRateLimiter; THROTTLED_PORTS_LEN]>,
    |         ------------------------------------------------------------- this field does not implement `Copy`
```

I suppose the alternative is to pull the array out of `Limits` somehow

Reply:
Hmm... `Limits` really shouldn't be `Copy` either... but that would technically be a breaking change to remove it... :frowning_face:. At any rate this should be:

```suggestion
                        let rate_limiter = &mut rate_limiter[port_index];
```